src/Microsoft.AspNetCore.Razor/Editor/ImplicitExpressionEditHandler.cs

```diff
@@ -149,10 +154,65 @@ private static bool IsSecondaryDotlessCommitInsertion(Span target, TextChange ch
                    change.OldLength == 0;
         }
 
-        private static bool IsAcceptableReplace(Span target, TextChange change)
+        private bool IsAcceptableReplace(Span target, TextChange change)
         {
             return IsEndReplace(target, change) ||
-                   (change.IsReplace && RemainingIsWhitespace(target, change));
+                (change.IsReplace && RemainingIsWhitespace(target, change));
+        }
+
+        private bool IsAcceptableIdentifierReplacement(Span target, TextChange change)
+        {
+            if (!change.IsReplace)
+            {
+                return false;
+            }
+
+            for (var i = 0; i < target.Symbols.Count; i++)
```

Comment:
Ehh, depends on your definition of small. I'd be shocked to see more than a few thousand. 

I thought about binary searching this but figured it'd be a premature optimization.